distribution/pull_v2.go

```diff
@@ -145,7 +147,7 @@ type v2LayerDescriptor struct {
 	src               distribution.Descriptor
 	ctx               context.Context
 	layerDownload     io.ReadCloser
-	downloadAttempts  uint8
+	downloadRetries   uint8 // Number of retries since last successful download attempt
```

Comment:
Why have a maximum number of times to retry the download?

Reply:
In principle, I dislike the idea of having goroutines that could conceivably run forever. But that's a good question and I'll need to think more about it.

We can for sure give more thought to our retry strategy (number of retry attempts, interval between retries, etc.).

Reply:
@lmbarros After reviewing upstream, they also have a `maxDownloadAttempts = 5`, so this seems okay to me.

Reply:
Hey guys, Rakshak here with Zenus - a client of yours.

I think it's expected that very long period of network outages can occur in the field. 
Our devices will connect to various networks of varying reliability/speed/etc.

Having the routine run forever seems completely reasonable on our end. I'd go as far to say that it is necessary.

Thank you

Reply:
Thanks @rakshak-t, this feedback is very useful! In fact, it would be great if you could expand a little bit -- and even better if you have and could share some actual numbers from your experience at Zenus. For example, how long are a typical and an exceptionally long network outages you see on the field?

> Is there anything I'm missing; is there a downside to having it run endlessly?

I am more worried about second-order effects. I am not sure this specific scenario is really possible, but here's an example. Say you have an image pull ongoing for days. Then, before it finishes, you release a new version which triggers a new image pull. Now that previous, unfinished pull will work against us, because it will be still consuming our limited network resources — and is likely to make our new update stall for even longer. I fear that running pulls endlessly might trigger this kind of scenario more frequently.

Reply:
@lmbarros So far we've only deployed a fraction of the number of devices that we're anticipating releasing in the coming months. Given the limited set of observations due to the above - the longest outage we've seen lasted about half of a day (8-14 hours). However, there could be circumstances of degraded network performance. This would be where a device is always running under a strained/weak network.
Having a slow trickle of an update that even takes days is better than it canceling after several hours to a day (which would lead to no update effectively).

This is just an outcome due to the large image sizes we deal with (several GB) coupled with the varied networks our devices can be connected to.

However, if that would truly result in second-order-instability, please let us know what you think a reasonable time out would be.

Thank you